### ISSUE TITLE
fix: pre hook script: Null check operator used on a null value

### ISSUE THREAD

**Issue by a-wallen** [VeryGoodOpenSource/dart_frog#1124]
**Description**

I'm running `dart_frog build` and hitting this error: 
```
An exception occurred while executing hook: C:\Users\awallen\AppData\Local\Mason\Cache\bundled\dart_frog_prod_server_0.1.0+1_dd63917756d9b4f8fd323148a45cb9d6c977ffb9\hooks\pre_gen.dart.
Error: Null check operator used on a null value
```

`pre_gen.dart:103`
I ran through the `pre_gen.dart` hook and it appears that the assertion is properly guarded 

```
    'globalMiddleware': configuration.globalMiddleware != null
        ? configuration.globalMiddleware!.toJson()
        : false,
```

**Steps To Reproduce**

1. Create a Dart Frog project:
```sh
dart_frog create sample
```

2. Adding an overridden dependency that has no matching main or dev dependency (from sample/):
```sh
dart pub add 'override:meta:'
```

3. Building a Dart Frog project (from sample/):
```sh
dart_frog build
```

4. See the error:

```txt
✓ Installing dependencies (0.4s)
✓ Bundling sources (13ms)
An exception occurred while executing hook: /Users/alestiago/.mason-cache/bundled/dart_frog_prod_server_0.1.0+1_e4edd283c6291e9406142f96559194db57c10097/hooks/pre_gen.dart.
Error: Null check operator used on a null value
```

You can access a sample with these changes at the [`dart_frog-1124-20240126`](https://github.com/alestiago/issues/tree/dart_frog-1124-20240126) branch.

**Expected Behavior**

Builds fine.

**Additional Context**

- Using `dart_frog: 1.0.1` with `flutter: 3.10.1`.
- Related to an external dependency issue (https://github.com/alexei-sintotski/pubspec_lock/issues/33)

**Comment by alestiago** [comment]
Hi @a-wallen thanks for opening an issue! Do you have a minimal code sample I can ran to reproduce the issue? 🙌

**Comment by a-wallen** [comment]
I'm using the API from the news toolkit. The dart dependency to `>=3.0.0 <4.0.0` and upgraded dart_frog to `1.0.1` IIRC

**Comment by alestiago** [comment]
What Dart Frog CLI version are you using? You can view your version using `dart_frog --version`

**Comment by a-wallen** [comment]
When I run into this again, i'll try again. I reverted back to 0.3.6 in the meantime to avoid this error

**Comment by a-wallen** [comment]
Ah if that's the case then you can just go ahead and close this issue. I was doing some strange things trying to get my news app running ASAP. It's likely that I'll be using the template to crank out more projects in the future. If I run into this again when I have a fresh clone of the template then I'll reopen the issue, but I don't have access to my dev PC right now and I don't want this to be in triage for that long. Thanks for the help.

**Comment by alestiago** [comment]
Thanks @a-wallen closing this for now, feel free to re-open!

**Comment by niklasbartsch** [comment]
I got the exactly same problem. I can't build with `>1.0.0` in one of my projects.

**Comment by alestiago** [comment]
Hi @niklasbartsch , do you have minimal reproductive steps for us to try and reproduce the issue?

**Comment by alestiago** [comment]
I think this issue might be related to an external dependency's [issue](https://github.com/alexei-sintotski/pubspec_lock/issues/33)  being used to locate [packages](https://github.com/VeryGoodOpenSource/dart_frog/blob/3132c16654d60cc9515f8ad1b6cd582928f1a5fb/bricks/dart_frog_prod_server/hooks/src/get_internal_path_dependencies.dart#L8).

I need to try and reproduce it, but it might be due to the use of "direct overridden" dependencies in the `pubsec.lock`.

**Comment by niklasbartsch** [comment]
I was not able to build a simple reproductive project.
The `dart_frog` version is `version: "1.0.1"`

**Comment by alestiago** [comment]
For those reading this, it seems to be an issue related to an external dependency happening when you define an overridenn dependency that is not actually overriding any dependency. In other words when using, `dependency_overrides` but there is no dependency with the same name in `dependencies` or `dev_dependencies`. Whilst this issue gets resolve, you can overcome it by manually adding the overridden dependency as a main or dev dependency.

**Comment by alestiago** [comment]
This should now be resolved, starting from Dart Frog CLI [v1.2.2](https://github.com/VeryGoodOpenSource/dart_frog/releases/tag/dart_frog_cli-v1.2.2).
